### ISSUE TITLE
Invalid location for object properties with ()

### ISSUE THREAD

**Issue by vjeux** [facebook/flow#3769]
Flow doesn't properly report the location as it doesn't include the last parenthesis. It messes up prettier empty line detection algorithm ( https://github.com/prettier/prettier/issues/1336 ).

```js
const Labels = {
  label1: (
    <fbt>
      Label 1
    </fbt>
  ),
  
  label2: (
    <fbt>
      Label 2
    </fbt>
  )
};
```

http://astexplorer.net/#/gist/3dea3bb4d1b4ae55846bc487bafe5664/4a4d4cbbed2e3445cd1bb863d3b7b910db25165b

![image](https://cloud.githubusercontent.com/assets/197597/25193632/9bf6038e-24ec-11e7-987d-9d64df5a0b32.png)

Also an issue for type casts

```js
o = {
  is_mounted: (undefined: ?boolean),
};
```

http://astexplorer.net/#/gist/3dea3bb4d1b4ae55846bc487bafe5664/de7ac7b309d8f18156d0224ff959ee847854dcb3

![image](https://cloud.githubusercontent.com/assets/197597/25193647/a851e8c8-24ec-11e7-8c3d-a6ded2ea2d0e.png)

**Comment by vjeux** [comment]
cc @alangpierce as you fixed a similar bug with #3745

**Comment by mroch** [comment]
coming right up
